examples/window-app/nrfconnect/main/include/AppTask.h

```diff
@@ -82,4 +86,8 @@ class AppTask
 #if CONFIG_CHIP_FACTORY_DATA
     chip::DeviceLayer::FactoryDataProvider<chip::DeviceLayer::InternalFlashFactoryData> mFactoryDataProvider;
 #endif
+
+#ifdef CONFIG_CHIP_ICD_SUBSCRIPTION_HANDLING
+    ICDUtil mICDUtils;
```

Comment:
If we always have only a single instance of `ICDUtil` why not making it static and move to the `util`? We could spare one #ifdef per sample. Maybe we could even move the whole `RegisterReadHandlerAppCallback()` there as well and have a single, argument-less call for apps.